From 75ae1a91dd112faef56bb4bf0459c87acc3c09d6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 20 May 2015 16:44:00 +0200
Subject: [PATCH] app: Fall back to window title instead of WM_CLASS

It's a bad fallback as it's clearly window-specific (rather than
app-specific), but it likely looks prettier when we fail to associate
a .desktop file ...
---
 src/shell-app.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shell-app.c b/src/shell-app.c
index dea610e..f1b71d9 100644
--- a/src/shell-app.c
+++ b/src/shell-app.c
@@ -420,7 +420,7 @@ shell_app_get_name (ShellApp *app)
       MetaWindow *window = window_backed_app_get_window (app);
       const char *name;
 
-      name = meta_window_get_wm_class (window);
+      name = meta_window_get_title (window);
       if (!name)
         name = C_("program", "Unknown");
       return name;
-- 
2.3.6

